File: LHDS.Core/Models/Foundations/OptOuts/OptOut.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using LHDS.Core.Models.Bases;

namespace LHDS.Core.Models.Foundations.OptOuts
{
    public class OptOut : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string NhsNumber { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string BatchReference { get; set; } = string.Empty;
        public string UniqueReference { get; set; } = string.Empty;
        public DateTimeOffset CacheTime { get; set; }
        public DateTimeOffset LastSentToMesh { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: LHDS.AdminPortal.Configurations/lhds.adminportal.configurations.client/src/models/optout/optout.ts
export class OptOut {
    public id: string;
    public nhsNumber: string;
    public status: string;
    public cacheTime: Date | undefined;
    public lastSentToMesh: Date | undefined;
    public uniqueReference?: string;
    public batchReference?: number;
    public createdDate?: Date | undefined;
    public createdBy?: string;
    public updatedDate?: Date | undefined;
    public updatedBy?: string;

    constructor(optout: OptOut) {
        this.id = optout.id ? optout.id : "";
        this.nhsNumber = optout.nhsNumber;
        this.status = optout.status || "";
        this.uniqueReference = optout.uniqueReference || "";
        this.batchReference = optout.batchReference;
        this.cacheTime = optout.cacheTime ? new Date(optout.cacheTime) : undefined;
        this.lastSentToMesh = optout.lastSentToMesh ? new Date(optout.lastSentToMesh) : undefined;
        this.createdDate = optout.createdDate ? new Date(optout.createdDate) : undefined;
        this.createdBy = optout.createdBy || "";
        this.updatedDate = optout.updatedDate ? new Date(optout.updatedDate) : undefined;
        this.updatedBy = optout.updatedBy || "";
    }
}